### PR TITLE
Add ApproxEq::abs_diff_{eq|ne} functions and macros

### LINKED ISSUE #8: Absolute tolerance
I find this library really useful, however sometimes you actually _want_ to compare two numbers using their absolute difference. Could it perhaps be desirable to have an additional macro for this purpose? I find this perhaps particularly useful when comparing with zero.

``` rust
// The name is perhaps not perfect
assert_absdiff_eq!(x, y, tol = 1e-8);

// I actually suggest not providing a default tolerance, as I believe
// absolute difference should only be used when you already have
// a specific tolerance in mind
```

I could certainly try to make a PR for this, but I thought I'd see if there's interest in it first.

### PR DESCRIPTION
**Pull request by brendanzab**
Closes #8.

The documentation could be a great deal better, especially given @Andlon's brilliantly in-depth answers on #8. We should definitely revisit it and give it the love it deserves. Other future things that might be interesting to explore is copying some design decisions from [the macros provided in rulinalg](https://docs.rs/rulinalg/0.4.2/rulinalg/macro.assert_matrix_eq.html).

### CODE REVIEW

**Comment by brendanzab** [comment]
Would like some feedback @Andolon! I just copied over the tests from `relative_eq`. I'm surprised more things didn't fail than just the infinities. Is this to be expected?

File: src/lib.rs [hunk 1]
```diff
@@ -192,6 +213,11 @@ macro_rules! impl_float_approx_eq {
             fn default_max_ulps() -> u32 { 4 }
 
             #[inline]
+            fn abs_diff_eq(&self, other: &$T, epsilon: $T) -> bool {
+                $T::abs(self - other) <= epsilon
```
Comment (Andlon) [inline src/lib.rs]: If this trait is intended to be possible to be implemented for unsigned integers, then you might run into underflow problems here. In `rulinalg` we handle this by a conditional on which element is bigger, see [here](https://github.com/AtheMathmo/rulinalg/blob/master/src/macros/matrix_eq.rs#L333). At the moment it's not a problem though

Comment (brendanzab) [inline src/lib.rs]: Yup, definitely.

**Comment by brendanzab** [comment]
Hmm, yes, I see. I guess it would be good to have some nice docs helping people understand how to choose the tolerance value. I don't really have enough experience to say though, and that was the biggest thing that stumped me with his article. I'm sure it would confuse other newcomers as well.

**Comment by brendanzab** [comment]
At the moment all functions have a default epsilon, so I'll stick with that. But it's definitely a good point that we should consider!

**Pull request merged**